### PR TITLE
Feature/portfolio read only - issue #443 #363

### LINKED ISSUE #363: Portfolio view is also showing deleted projects
**Describe the bug**
As a member of the group "Portføjeinnsyn" you will be able to see all the projects in the portfolio view. 
The bug in this instance is that you see deleted projects as well, meaning deleted sites is still visible.

**To Reproduce**

1. Go to https://prosjektportalen.sharepoint.com/sites/pp365/SitePages/Portef%C3%B8ljeoversikt.aspx as prosjektportalen admin user.
2. Look at the project overview list and find e.g "Bitpro demoprosjekt"
3. Go to https://prosjektportalen-admin.sharepoint.com/_layouts/15/online/AdminHome.aspx#/recycleBin as prosjektportalen admin user
4. Find "Bitpro demoprosjekt" in the deleted sites section
5. EXPERIENCE, VITNESS & LIVE THE BUG!

**Expected behavior**
I would like deleted project not to be visible in the portfolio view

**Screenshots**

**About your setup**

- Installed version [found under {Prosjektportalen}/Lists/Installasjonslogg/AllItems.aspx]

**Additional context**

### LINKED ISSUE #443: Read-only access to portfolio frontpage project view
**Is your feature request related to a problem? Please describe.**
In some instances you might want to give some users read access to all the project cards on the frontpage. This is hard to achieve today without giving the users read-access or member-access to all projects. Similar to what was done in #243 / #305 

**Describe the solution you'd like**
If someone is in the group "Porteføljeinnsyn" they will be able to view all projects in the portfolio frontpage project view. User is not able to enter the project, only view the cards

Related to ASSIST-3389 (16h)

### PR DESCRIPTION
**Pull request by Rundez**
### Your checklist for this pull request

- [X] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your main!
- [X] Make sure you are making a pull request against the **dev** branch (left side). Also you should start *your branch* off *dev*.
- [X] Check the commit's or even all commits' message 
- [X] Check if your code additions will fail linting checks
- [X] Remember: Add PR description to [CHANGELOG](https://github.com/Puzzlepart/prosjektportalen365/blob/dev/CHANGELOG.md) with the ID that matches this PR

### Description

Added the possibility to see all projects when the user is in the "Porteføljeinnsyn" group. This can be toggled on and off in the projectlist web part. If it is toggled on, the user(s) in "Porteføljeinnsyn" will notice additional buttons appear right of the search bar. These buttons will change the current view, toggling between seeing all projects, and projects that the user is a member of. 

Also added a function to filter deleted projects in "Porteføljeoversikt", as it was related to this task. No deleted projects will now appear in the table. 

### How to test
**Portfolio view**
1. Add a user to the "Porteføljeinnsyn" group. 
2. Navigate to the project list property pane and turn on "Vis alle prosjekter"
3. Toggle the buttons that appear. No deleted projects should appear, and projects you are not a member of should not ble clickable. 

**Portfolio overview**
1. Add a user to the "Porteføljeinnsyn" group.
2. Navigate to "Porteføljeoversikt" 
3. All projects except for deleted ones should be displayed.

### Relevant issues (if applicable)
fixes #363 
fixes #443 

💔Thank you!

### CODE REVIEW

File: SharePointFramework/PortfolioWebParts/src/data/index.ts [hunk 1]
```diff
@@ -203,19 +204,40 @@ export class DataAdapter {
         siteId,
         siteIdProperty
       )
+      const projectList = await sp.web.lists
```
Comment (tarjeieo) [inline SharePointFramework/PortfolioWebParts/src/data/index.ts:207]: This doesn't scale well (beyond 500 items), and is also hardcoded wrt. "Avsluttet".

**Comment by olemp** [comment]
It's a lot going on in this PR. I see that it's trying to solve two issues. Would it better to split it into two PRs? 

cc @tarjeieo @Rundez

File: SharePointFramework/PortfolioWebParts/src/data/index.ts (hunk 1 above)
Comment (olemp) [reply SharePointFramework/PortfolioWebParts/src/data/index.ts]: +1

**Comment by Rundez** [comment]
Yes, indeed. Both issues have grown quite a bit in the implementation. I have talked to Tarjei about some further changes, so we can postpone this PR for some time. You might be right about splitting, the issues are not dependent on each other, so it would work out.

**Comment by olemp** [comment]
> Yes, indeed. Both issues have grown quite a bit in the implementation. I have talked to Tarjei about some further changes, so we can postpone this PR for some time. You might be right about splitting, the issues are not strictly dependent on each other as I first thought.
> 
> cc @tarjeieo @olemp

Let's close it for now.

**Pull request closed without merging**